### ISSUE TITLE
Check the available system space before starting the download.

### ISSUE THREAD

**Issue by setraj** [arvyshka/blazer#32]
Probably, prompt the user with available space and confirm to go ahead with the download or not.

**Comment by arvyshka** [comment]
Yes good, can we show this prompt if there will be less space ? I haven't seen other downloaders like wget and curl asking for every download

**Comment by setraj** [comment]
Say, the file size is 5Gb and the available space is 2.5 Gb. 
do we want to stop the download in between?  wouldn't it be better to prompt the user beforehand?

**Comment by arvyshka** [comment]
@setraj Yes, we are in the same agreement basically, I was talking about this case:

the required size is already there in the disk, should we still prompt ? or go ahead ?

**Comment by setraj** [comment]
ok, I get you. 
Yeah in that case we can silently proceed with the download.
in case the space is not sufficient, we can notify the user that the download can't be continued because of insufficient space.

**Comment by arvyshka** [comment]
ok please proceed implementation
